src/daemon.rs

```diff
@@ -621,6 +641,25 @@ impl Daemon {
             .chain_err(|| "invalid testmempoolaccept reply")
     }
 
+    pub fn submit_package(
+        &self,
+        txhex: Vec<String>,
+        maxfeerate: Option<f64>,
+        maxburnamount: Option<f64>,
+    ) -> Result<SubmitPackageResult> {
+        let params = match (maxfeerate, maxburnamount) {
+            (Some(rate), Some(burn)) => {
+                json!([txhex, format!("{:.8}", rate), format!("{:.8}", burn)])
+            }
+            (Some(rate), None) => json!([txhex, format!("{:.8}", rate)]),
+            (None, Some(burn)) => json!([txhex, format!("{:.8}", burn)]),
```

Comment:
The RPC accepts the following:
`submitpackage ["rawtx",...] ( maxfeerate maxburnamount )`

So in the `(None, Some(burn))` branch you are passing in `burn` where `maxfeerate` should go.

Option 1: The default for `maxfeerate` is `0.10` so we can use that (as a string `"0.10"`). This will be a bad choice if Core ever changes this default.

Option 2: I'd assume that we need to write `null` in the final JSON. (Pretty sure `None::<()>` should get the job done, might want to check if Bitcoin Core accepts `null` in this position)

Reply:
Looks like the `json!` macro views the inner `[]` as a part of the macro, so elements of this "array" don't have to be the same type (unlike Rust arrays).

This compiles:

```rust
(None, Some(burn)) => json!([txhex, null, format!("{:.8}", burn)]),
```

But the question is whether Core will accept null in this position in the final JSON params.

Reply:
It seems Core accepts null as valid parameter: 
```
bitcoin-cli -testnet4 submitpackage '["030000000001016c7f7a9a9b02b8fa09c5392e77fedd12f5646ba23bb7c9ba6236ac44849255f90000000000fdffffff02e0990300000000001600141813c9277780a47d99ebe14158d66dcfab2ae551e803000000000000116a0f5468697320697320696c6c6567616c024730440220770629b7383444e94b1ce40bea0c16909689b22fd8de9cb1dcb0330ed235b45402204fcdb88102a871f00936fd5508c43036b5fc6e51ddc2af17f32ca179b2f7ad8f012102b0f50fb9e58f037212a4cb6c654d297de64068cd1b9473d28f27ee07612f988a00000000","03000000000101aef748331067801744fef4749514ca2b69c844e5270f7be5c8723c90310b3ce90000000000fdffffff01f8950300000000001600141813c9277780a47d99ebe14158d66dcfab2ae55102473044022030b6fc0e894890c53650330bf899c776d46c41add8b548b92b4080dc074faf63022017dfbbbfb5df20bd9c76b85e603e598f67b0f5eac04870f3b0551d5cd6850ee7012102b0f50fb9e58f037212a4cb6c654d297de64068cd1b9473d28f27ee07612f988a00000000"]'         
error code: -25
error message:
Unspendable output exceeds maximum configured by user (maxburnamount)
bitcoin-cli -testnet4 submitpackage '["030000000001016c7f7a9a9b02b8fa09c5392e77fedd12f5646ba23bb7c9ba6236ac44849255f90000000000fdffffff02e0990300000000001600141813c9277780a47d99ebe14158d66dcfab2ae551e803000000000000116a0f5468697320697320696c6c6567616c024730440220770629b7383444e94b1ce40bea0c16909689b22fd8de9cb1dcb0330ed235b45402204fcdb88102a871f00936fd5508c43036b5fc6e51ddc2af17f32ca179b2f7ad8f012102b0f50fb9e58f037212a4cb6c654d297de64068cd1b9473d28f27ee07612f988a00000000","03000000000101aef748331067801744fef4749514ca2b69c844e5270f7be5c8723c90310b3ce90000000000fdffffff01f8950300000000001600141813c9277780a47d99ebe14158d66dcfab2ae55102473044022030b6fc0e894890c53650330bf899c776d46c41add8b548b92b4080dc074faf63022017dfbbbfb5df20bd9c76b85e603e598f67b0f5eac04870f3b0551d5cd6850ee7012102b0f50fb9e58f037212a4cb6c654d297de64068cd1b9473d28f27ee07612f988a00000000"]' null 0.00001000
{
  "package_msg": "success",
  "tx-results": {
    "89ea4e6cf5b512181ccf9ef2fce6232ed93426af4f6ff34b5401414205620d5d": {
      "txid": "e93c0b31903c72c8e57b0f27e544c8692bca149574f4fe44178067103348f7ae",
      "vsize": 136,
      "fees": {
        "base": 0.00000000,
        "effective-feerate": 0.00004065,
        "effective-includes": [
          "89ea4e6cf5b512181ccf9ef2fce6232ed93426af4f6ff34b5401414205620d5d",
          "3d1e59c97eb350d6e15ed57de07de57623715b0e2f915dcec9ee8ddd970f0a16"
        ]
      }
    },
    "3d1e59c97eb350d6e15ed57de07de57623715b0e2f915dcec9ee8ddd970f0a16": {
      "txid": "1545ac2cc3448aae985e41582d30bca8a2afd08b212a0de58d2e36516545a059",
      "vsize": 110,
      "fees": {
        "base": 0.00001000,
        "effective-feerate": 0.00004065,
        "effective-includes": [
          "89ea4e6cf5b512181ccf9ef2fce6232ed93426af4f6ff34b5401414205620d5d",
          "3d1e59c97eb350d6e15ed57de07de57623715b0e2f915dcec9ee8ddd970f0a16"
        ]
      }
    }
  },
  "replaced-transactions": [
  ]
}